nativelib/src/main/scala/java/lang/Object.scala

```diff
@@ -4,34 +4,40 @@ import scala.scalanative.native._
 import scala.scalanative.runtime
 
 class _Object {
-  def _equals(that: _Object): scala.Boolean =
-    this.cast[Word] == that.cast[Word]
+  @inline def __equals(that: _Object): scala.Boolean =
+    this eq that
 
-  def _hashCode(): scala.Int =
+  @inline def __hashCode(): scala.Int =
     this.cast[Word].hashCode
 
-  def _toString(): String =
+  @inline def __toString(): String =
     getClass.getName + "@" + Integer.toHexString(hashCode)
 
-  def _getClass(): _Class[_] =
+  @inline def __getClass(): _Class[_] =
     new _Class(runtime.getType(this))
 
-  def _notify(): Unit =
+  @inline def __notify(): Unit =
     runtime.getMonitor(this)._notify
 
-  def _notifyAll(): Unit =
+  @inline def __notifyAll(): Unit =
     runtime.getMonitor(this)._notifyAll
 
-  def _wait(): Unit =
+  @inline def __wait(): Unit =
     runtime.getMonitor(this)._wait
 
-  def _wait(timeout: scala.Long): Unit =
+  @inline def __wait(timeout: scala.Long): Unit =
     runtime.getMonitor(this)._wait(timeout)
 
-  def _wait(timeout: scala.Long, nanos: Int): Unit =
+  @inline def __wait(timeout: scala.Long, nanos: Int): Unit =
     runtime.getMonitor(this)._wait(timeout, nanos)
 
-  protected def _clone(): _Object = ???
+  @inline def __scala_==(other: _Object): scala.Boolean =
+    if (this eq null) other eq null else __equals(other)
```

Comment:
`this eq null`, really!?

How does this work with the dynamic dispatch that would have to happen on `(null: Any).__scala_==(that)`?